File: emby-lib/src/main/java/us/nineworlds/serenity/emby/model/Directory.java
package us.nineworlds.serenity.emby.model;

import androidx.annotation.Nullable;
import java.util.List;
import us.nineworlds.serenity.common.media.model.IDirectory;
import us.nineworlds.serenity.common.media.model.IGenre;
import us.nineworlds.serenity.common.media.model.ILocation;

public class Directory implements IDirectory {
  private String thumb;
  private String banner;
  private int secondary;
  private List<IGenre> generes;
  private String ratingKey;
  private String studio;
  private String rating;
  private String year;
  private String contentRating;
  private String summary;
  private String leafCount;
  private String viewedLeafCount;
  private String title;
  private String art;
  private int refreshing;
  private String type;
  private String agent;
  private String scanner;
  private String language;
  private String uuid;
  private long updatedAt;
  private long createdAt;
  private String prompt;
  private String search;
  private List<ILocation> locations;
  private String key;

  @Nullable @Override public String getThumb() {
    return thumb;
  }

  @Override public void setThumb(@Nullable String thumb) {
    this.thumb = thumb;
  }

  @Nullable @Override public String getBanner() {
    return banner;
  }

  @Override public void setBanner(@Nullable String banner) {
    this.banner = banner;
  }

  @Override public int getSecondary() {
    return secondary;
  }

  @Override public void setSecondary(int secondary) {
    this.secondary = secondary;
  }

  @Override public List<IGenre> getGenres() {
    return generes;
  }

  @Override public void setGenres(List<IGenre> genres) {
    this.generes = genres;
  }

  @Nullable @Override public String getRatingKey() {
    return ratingKey;
  }

  @Override public void setRatingKey(@Nullable String ratingKey) {
    this.ratingKey = ratingKey;
  }

  @Override public String getStudio() {
    return studio;
  }

  @Override public void setStudio(@Nullable String studio) {
    this.studio = studio;
  }

  @Override public String getRating() {
    return rating;
  }

  @Override public void setRating(@Nullable String rating) {
    this.rating = rating;
  }

  @Override public String getYear() {
    return year;
  }

  @Override public void setYear(@Nullable String year) {
    this.year = year;
  }

  @Nullable @Override public String getContentRating() {
    return contentRating;
  }

  @Override public void setContentRating(@Nullable String contentRating) {
    this.contentRating = contentRating;
  }

  @Override public String getSummary() {
    return summary;
  }

  @Override public void setSummary(@Nullable String summary) {
    this.summary = summary;
  }

  @Nullable @Override public String getLeafCount() {
    return leafCount;
  }

  @Override public void setLeafCount(@Nullable String leafCount) {
    this.leafCount = leafCount;
  }

  @Override public String getViewedLeafCount() {
    return viewedLeafCount;
  }

  @Override public void setViewedLeafCount(String viewedLeafCount) {
    this.viewedLeafCount = viewedLeafCount;
  }

  @Nullable @Override public String getTitle() {
    return title;
  }

  @Override public void setTitle(@Nullable String title) {
    this.title = title;
  }

  @Nullable @Override public String getArt() {
    return art;
  }

  @Override public void setArt(@Nullable String art) {
    this.art = art;
  }

  @Override public int getRefreshing() {
    return refreshing;
  }

  @Override public void setRefreshing(int refreshing) {
    this.refreshing = refreshing;
  }

  @Nullable @Override public String getType() {
    return type;
  }

  @Override public void setType(@Nullable String type) {
    this.type = type;
  }

  @Nullable @Override public String getAgent() {
    return agent;
  }

  @Override public void setAgent(@Nullable String agent) {
    this.agent = agent;
  }

  @Nullable @Override public String getScanner() {
    return scanner;
  }

  @Override public void setScanner(@Nullable String scanner) {
    this.scanner = scanner;
  }

  @Nullable @Override public String getLanguage() {
    return language;
  }

  @Override public void setLanguage(@Nullable String language) {
    this.language = language;
  }

  @Nullable @Override public String getUuid() {
    return uuid;
  }

  @Override public void setUuid(@Nullable String uuid) {
    this.uuid = uuid;
  }

  @Override public long getUpdatedAt() {
    return updatedAt;
  }

  @Override public void setUpdatedAt(long updatedAt) {
    this.updatedAt = updatedAt;
  }

  @Override public long getCreatedAt() {
    return createdAt;
  }

  @Override public void setCreatedAt(long createdAt) {
    this.createdAt = createdAt;
  }

  @Nullable @Override public List<ILocation> getLocations() {
    return locations;
  }

  @Override public void setLocation(@Nullable  List<ILocation> location) {
    this.locations = location;
  }

  @Nullable @Override public String getPrompt() {
    return prompt;
  }

  @Override public void setPrompt(@Nullable String prompt) {
    this.prompt = prompt;
  }

  @Nullable @Override public String getSearch() {
    return search;
  }

  @Override public void setSearch(String search) {
    this.search = search;
  }

  @Override public void setLocations(List<ILocation> locations) {
    this.locations = locations;
  }

  @Override public String getKey() {
    return key;
  }

  @Override public void setKey(String key) {
    this.key = key;
  }
}

File: jellyfin-lib/src/main/kotlin/us/nineworlds/serenity/jellyfin/model/Directory.kt
package us.nineworlds.serenity.jellyfin.model

import us.nineworlds.serenity.common.media.model.IDirectory
import us.nineworlds.serenity.common.media.model.IGenre
import us.nineworlds.serenity.common.media.model.ILocation

class Directory : IDirectory {
    private var _thumb: String? = null
    private var _banner: String? = null
    private var _secondary: Int = 0
    private var _genres: List<IGenre>? = null
    private var _ratingKey: String? = null
    private var _studio: String? = null
    private var _rating: String? = null
    private var _year: String? = null
    private var _contentRating: String? = null
    private var _summary: String? = null
    private var _leafCount: String? = null
    private var _viewedLeafCount: String? = null
    private var _title: String? = null
    private var _art: String? = null
    private var _refreshing: Int = 0
    private var _type: String? = null
    private var _agent: String? = null
    private var _scanner: String? = null
    private var _language: String? = null
    private var _uuid: String? = null
    private var _updatedAt: Long = 0
    private var _createdAt: Long = 0
    private var _prompt: String? = null
    private var _search: String? = null
    private var _locations: List<ILocation>? = null
    private var _key: String? = null

    override fun getThumb(): String? = _thumb
    override fun setThumb(thumb: String?) {
        _thumb = thumb
    }

    override fun getBanner(): String? = _banner
    override fun setBanner(banner: String?) {
        _banner = banner
    }

    override fun getSecondary(): Int = _secondary
    override fun setSecondary(secondary: Int) {
        _secondary = secondary
    }

    override fun getGenres(): List<IGenre>? = _genres
    override fun setGenres(genres: List<IGenre>?) {
        _genres = genres
    }

    override fun getRatingKey(): String? = _ratingKey
    override fun setRatingKey(ratingKey: String?) {
        _ratingKey = ratingKey
    }

    override fun getStudio(): String? = _studio
    override fun setStudio(studio: String?) {
        _studio = studio
    }

    override fun getRating(): String? = _rating
    override fun setRating(rating: String?) {
        _rating = rating
    }

    override fun getYear(): String? = _year
    override fun setYear(year: String?) {
        _year = year
    }

    override fun getContentRating(): String? = _contentRating
    override fun setContentRating(contentRating: String?) {
        _contentRating = contentRating
    }

    override fun getSummary(): String? = _summary
    override fun setSummary(summary: String?) {
        _summary = summary
    }

    override fun getLeafCount(): String? = _leafCount
    override fun setLeafCount(leafCount: String?) {
        _leafCount = leafCount
    }

    override fun getViewedLeafCount(): String? = _viewedLeafCount
    override fun setViewedLeafCount(viewedLeafCount: String?) {
        this._viewedLeafCount = viewedLeafCount
    }

    override fun getTitle(): String? = _title
    override fun setTitle(title: String?) {
        _title = title
    }

    override fun getArt(): String? = _art
    override fun setArt(art: String?) {
        _art = art
    }

    override fun getRefreshing(): Int = _refreshing
    override fun setRefreshing(refreshing: Int) {
        _refreshing = refreshing
    }

    override fun getType(): String? = _type
    override fun setType(type: String?) {
        _type = type
    }

    override fun getAgent(): String? = _agent
    override fun setAgent(agent: String?) {
        _agent = agent
    }

    override fun getScanner(): String? = _scanner
    override fun setScanner(scanner: String?) {
        _scanner = scanner
    }

    override fun getLanguage(): String? = _language
    override fun setLanguage(language: String?) {
        _language = language
    }

    override fun getUuid(): String? = _uuid
    override fun setUuid(uuid: String?) {
        _uuid = uuid
    }

    override fun getUpdatedAt(): Long = _updatedAt
    override fun setUpdatedAt(updatedAt: Long) {
        _updatedAt = updatedAt
    }

    override fun getCreatedAt(): Long = _createdAt
    override fun setCreatedAt(createdAt: Long) {
        _createdAt = createdAt
    }

    override fun getLocations(): List<ILocation>? = _locations
    override fun setLocation(location: List<ILocation>?) {
        _locations = location
    }

    override fun getPrompt(): String? = _prompt
    override fun setPrompt(prompt: String?) {
        _prompt = prompt
    }

    override fun getSearch(): String? = _search
    override fun setSearch(search: String?) {
        this._search = search
    }

    override fun setLocations(locations: List<ILocation>?) {
        _locations = locations
    }

    override fun getKey(): String? = _key
    override fun setKey(key: String?) {
        _key = key
    }
}